tests/imputers/test_BaseImputer.py

```diff
@@ -2,202 +2,268 @@
 
 import numpy as np
 import pandas as pd
+import polars as pl
 import pytest
-import test_aide as ta
 from sklearn.exceptions import NotFittedError
 
 import tests.test_data as d
+from tests import utils as u
 from tests.base_tests import (
     ColumnStrListInitTests,
     GenericFitTests,
     OtherBaseBehaviourTests,
 )
 
 
+@pytest.fixture
+def library(request):
+    return request.param
+
+
+@pytest.fixture()
+def test_fit_df(request):
+    library = getattr(request, "param", "pandas")
+    df = pd.DataFrame(
+        {
+            "a": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0],
+            "b": ["a", "b", "c", "d", "e", "f", np.nan],
+            "c": ["a", "b", "c", "d", "e", "f", np.nan],
+        },
+    )
+    if library == "polars":
+        return pl.from_pandas(df)
```

Comment:
Q: if this is our approach to producing expected polars df (i.e converting form pandas) then maybe we can wrap the conversion in comparison logic? Had a bit of a discussion on prods/cons of this approach here https://github.com/lvgig/tubular/pull/328#issuecomment-2414138420 

Tentative conclusion there was we should probably explicitly define each data frame.

Have since then considered trying a df comparison function that internally uses nw.from_native on both dfs and compares using pl.assert_frame_equal?  I really like the idea of getting to a place where we only need define one set of expected values, but I'm honestly very torn between the safety of explicitly defining both pandas and pl dfs and something like this to keep bloat down.  Keen to have others thoughts on this.  @limlam96 @adamsardar?

Reply:
just off a call with David, I think what we've settled on is where we have data fixtures/functions like: 
```
@pytest.fixture()
def test_fit_df(request):
    library = getattr(request, "param", "pandas")
    df = pd.DataFrame(...)
    if library == "polars":
        return pl.from_pandas(df)
    return df
```

We change to like:
```
@pytest.fixture()
def test_fit_df(request):
    library = getattr(request, "param", "pandas")
    dataframe_init=tests.utils.get_dataframe_init(library)
    df = dataframe_init(...)
    return df
```
and get_dataframe_init is a new function that returns pl.DataFrame or pd.DataFrame depending on the input library argument

hope that makes sense @alphanumericmale ? apologies for the change in approach

Reply:
```
library = getattr(request, "param", "pandas") 
dataframe_init=tests.utils.get_dataframe_init(library)
```

This feels like deep magick on the Python side - worth wrapping in some sugar methods so that it's a little more obvious for new contributors. Happy to kick into another ticket as this is scope creep.

Reply:
https://github.com/lvgig/tubular/issues/338